src/flamenco/runtime/fd_blockstore.h

```diff
@@ -272,6 +272,35 @@ typedef struct fd_block_idx fd_block_idx_t;
 #define MAP_KEY_HASH(key) ((uint)(key)) /* finalized slots are guaranteed to be unique so perfect hashing */
 #include "../../util/tmpl/fd_map_dynamic.c"
 
+/* fd_blockstore_ser is a serialization context for archiving a block to
+   disk. */
+
+struct fd_blockstore_ser {
+  fd_block_map_t * block_map;
+  fd_block_t     * block;
+  uchar          * data;
+};
+typedef struct fd_blockstore_ser fd_blockstore_ser_t;
+
+/* Archives a block and block map entry to fd */
+ulong
+fd_blockstore_block_archive( fd_blockstore_ser_t * ser, int fd, ulong slot );
```

Comment:
sorry, i realized `checkpt` is the more consistent name w/ other parts of the code